src/Raven.Server/Documents/Subscriptions/SubscriptionFetcher.cs

```diff
@@ -35,9 +40,12 @@ public IEnumerable<T> GetEnumerator()
             }
 
             FetchingFrom = FetchingOrigin.Storage;
-            foreach (var item in FetchByEtag())
+            using (var fetchByEtagEnumerator = FetchByEtag())
```

Comment:
my bad, didn't know that you can execute foreach on enumerator (since C# 9). Now we have a double dispose. One in foreach and 2nd one here.